### PR TITLE
Folder support

### PR DESCRIPTION
**Pull request by MarkusHorstmann**

### CODE REVIEW

**Comment by MarkusHorstmann** [comment]
@scoxen1 Can you take a quick look at the front end changes, and maybe the ancestry/breadcrump code?

File: api/CESMII.ProfileDesigner.DAL/Utils/ProfileMapperUtil.cs
Comment (scoxen1) [inline api/CESMII.ProfileDesigner.DAL/Utils/ProfileMapperUtil.cs:1]: TODO: Can we move the Folder type node id contstant to appSettings.json

Comment (MarkusHorstmann) [reply api/CESMII.ProfileDesigner.DAL/Utils/ProfileMapperUtil.cs:1]: This is not really a setting. The Folder Type nodeid is cast in stone for eternity, by the OPC spec. We could make it a static const string in the .cs file?

File: frontend/src/services/AttributesService.js [hunk 1]
```diff
@@ -371,8 +371,6 @@ export const onChangeCompositionShared = (match, item) => {
         item.composition.intermediateObjectId = intermediateObjectId;
         item.composition.intermediateObjectName = intermediateObjectName;
         //copy some values into the profile_composition data record
-        item.browseName = match.browseName; //this becomes critical for adding on server side in new scenario
```
Comment (MarkusHorstmann) [inline frontend/src/services/AttributesService.js:374]: Need to make sure this doesn't regress anything.

File: api/CESMII.ProfileDesigner.DAL/ProfileTypeDefinitionDAL.cs [hunk 2]
```diff
@@ -223,7 +223,7 @@ private static bool MatchIdentity(int? entityId, string entityOpcNodeId, string
             }
             else
             {
-                return modelOpcNodeId == entityOpcNodeId && modelNamespace == entityNamespace;
+                return (modelOpcNodeId != null || entityOpcNodeId != null) && modelOpcNodeId == entityOpcNodeId && modelNamespace == entityNamespace;
```
Comment (MarkusHorstmann) [inline api/CESMII.ProfileDesigner.DAL/ProfileTypeDefinitionDAL.cs:226]: Discussion on the browsename change in attributeservice.js: I think this change makes the absence of a browsename work: before we would latch on the first empty/null browsename.

File: frontend/src/services/AttributesService.js (hunk 1 above)
Comment (MarkusHorstmann) [reply frontend/src/services/AttributesService.js:374]: See other comment: https://github.com/cesmii/ProfileDesigner/pull/93#discussion_r1206062384. This is indeed no longer necessary.

**Pull request merged**